### PR TITLE
Improving error message on autoUpdate mismatch

### PR DESCRIPTION
**Pull request by kritika-singh3**
Issue: #6031

Description:
 - showing the list of pipelines which uses the material in question (irrespective of autoUpdate value)
 - also showing the id of the config repository - if used

### CODE REVIEW

**Comment by kritika-singh3** [comment]
Preview:

Editing material:

Screen Shot 2020-02-05 at 1 03 14 PM

Screen Shot 2020-02-05 at 1 03 58 PM

Adding config-repo for a material which has `autoUpdate` set to `false`:

Screen Shot 2020-02-05 at 1 04 50 PM

**Comment by arvindsv** [comment]
I know the message is already big, but it is unclear exactly what is happening. For instance, it talks about `autoUpdate` attribute. Maybe we can clarify that it is "Poll for new changes" as well.

How about: 

```
This material is used elsewhere with a different value for autoUpdate ("Poll for changes"). Those values should be the same. Other places it is used in are:

This pipeline: Auto update disabled
Config repository "sdsd": Auto update enabled
Pipeline "up42": Auto update enabled
```

... or something similar?

**Comment by arvindsv** [comment]
This material is used elsewhere with a different value for autoUpdate ("Poll for changes"). Those values should be the same. This materail: auto update enabled. Pipelines: up42 (auto update disabled), up43 (auto update disabled). Config repository: sdsd (auto update disabled).

**Comment by kritika-singh3** [comment]
Update message:

Screen Shot 2020-02-05 at 5 15 20 PM

Screen Shot 2020-02-05 at 5 11 20 PM

File: config/config-api/src/main/java/com/thoughtworks/go/config/ConfigSaveValidationContext.java [hunk 1]
```diff
@@ -283,6 +284,19 @@ public MaterialConfigs getAllMaterialsByFingerPrint(String fingerprint) {
         return matchingMaterials == null ? new MaterialConfigs() : matchingMaterials;
     }
 
+    @Override
+    public Map<CaseInsensitiveString, Boolean> getPipelinesWithMaterial(String fingerprint) {
```
Comment (bdpiparva) [inline config/config-api/src/main/java/com/thoughtworks/go/config/ConfigSaveValidationContext.java]: Don't you think method name is mis-leading?

File: config/config-api/src/main/java/com/thoughtworks/go/config/ValidationContext.java [hunk 2]
```diff
@@ -84,5 +88,9 @@ default SystemEnvironment systemEnvironment() {
     default PolicyValidationContext getPolicyValidationContext() {
         return null;
     }
+
+    default Map<CaseInsensitiveString, Boolean> getPipelinesWithMaterial(String fingerprint) {
```
Comment (bdpiparva) [inline config/config-api/src/main/java/com/thoughtworks/go/config/ValidationContext.java]: Same as above

Comment (kritika-singh3) [reply config/config-api/src/main/java/com/thoughtworks/go/config/ValidationContext.java]: Couldn't think of a better name. Any suggestions?

File: config/config-api/src/main/java/com/thoughtworks/go/config/ConfigSaveValidationContext.java (hunk 1 above)
Comment (kritika-singh3) [reply config/config-api/src/main/java/com/thoughtworks/go/config/ConfigSaveValidationContext.java]: This is the implementation of the method in `ValidationContext`

File: config/config-api/src/main/java/com/thoughtworks/go/config/ValidationContext.java (hunk 2 above)
Comment (kritika-singh3) [reply config/config-api/src/main/java/com/thoughtworks/go/config/ValidationContext.java]: Updated to `getPipelineToMaterialAutoUpdateMapByFingerprint`

**Comment by maheshp** [comment]
@kritika-singh3 there are couple of things which needs to be looked at,
1. This error does not show up when you try to add a new pipeline, there is just a generic validation error.
2. The usages of the material comes as part of the error string and I think it's hard to read for UX perspective. I understand that the entire error message is returned as a single string, so you think we can do something to show the usages on a separate line.

**Comment by maheshp** [comment]
> This error does not show up when you try to add a new pipeline, there is just a generic validation error.

There is a PR https://github.com/gocd/gocd/pull/7859 to fix the issue on the add pipeline wizard.

> The usages of the material comes as part of the error string and I think it's hard to read for UX perspective. I understand that the entire error message is returned as a single string, so you think we can do something to show the usages on a separate line.

Fixed by - https://github.com/gocd/gocd/pull/7849

**Pull request merged**